agent/agent-bootstrap/src/main/java/com/microsoft/applicationinsights/agent/bootstrap/configuration/ConfigurationBuilder.java

```diff
@@ -57,31 +63,45 @@
     private static final List<ConfigurationMessage> configurationMessages = new CopyOnWriteArrayList<>();
 
     public static Configuration create(Path agentJarPath) throws IOException {
-
         Configuration config = loadConfigurationFile(agentJarPath);
-        PreviewConfiguration preview = config.instrumentationSettings.preview;
+        overlayEnvVars(config);
 
-        preview.roleName = overlayWithEnvVar(APPLICATIONINSIGHTS_ROLE_NAME, WEBSITE_SITE_NAME, preview.roleName);
-        preview.roleInstance =
-                overlayWithEnvVar(APPLICATIONINSIGHTS_ROLE_INSTANCE, WEBSITE_INSTANCE_ID, preview.roleInstance);
+        return config;
+    }
+
+    private static void loadLogCaptureEnvVar(PreviewConfiguration preview) {
+        if (preview.instrumentation.containsKey("logging")) {
+            Map<String, Object> logging = preview.instrumentation.get("logging");
+            logging.put("threshold", overlayWithEnvVar(APPLICATIONINSIGHTS_LOG_CAPTURE_THRESHOLD, (String) logging.get("threshold")));
+        }
```

Comment:
i don't think the condition is needed now with the new initialization? (also we always want to overlay env var)

```suggestion
        Map<String, Object> logging = preview.instrumentation.get("logging");
        logging.put("threshold", overlayWithEnvVar(APPLICATIONINSIGHTS_LOG_CAPTURE_THRESHOLD, (String) logging.get("threshold")));
```

Reply:
The new initialization caused the micrometer to fail.  micrometer is using the same preview.instrumentation.. we can't initialize it to logging only.  it also supports for any kind of instrumentation names.  I need to delete the initialization and revert back to what i had originally.

Reply:
ok, makes sense

can you add unit tests that cover micrometer also? will be a good way to verify configuration behavior

also, need to overlay env var whether or not `preview.instrumentation.containsKey("logging")`, can you add unit test for that also?